### PR TITLE
Fixed infinity scroll on eco news page

### PR DESCRIPTION
**Pull request by RomanDeveloperAcc**
Fixed infinity scroll on eco news page:
-fixed scroll without filters;
-fixed scroll with filters;
-fixed 'items found' counter;

### CODE REVIEW

File: src/app/component/eco-news/components/news-list/news-list.component.ts [hunk 1]
```diff
@@ -55,46 +54,43 @@ export class NewsListComponent implements OnInit, OnDestroy {
...
-    this.allEcoNews.forEach((element, index) => {
-      if (index >= this.iterator && index - this.iterator < loadingLength) {
-        this.elements[index] = element;
-      }
-    });
-
-    this.iterator = this.elements.length;
+    if (this.gridOutput) {
+      this.ecoNewsSubscription = this.ecoNewsService.getNewsListByTags(this.iterator++, 12, this.gridOutput)
+        .subscribe(this.setList.bind(this));
```
Comment (mplesha) [inline src/app/component/eco-news/components/news-list/news-list.component.ts:64]: The method 'bind' hasn't been used in Angular for a long time. Please check - is it really necessary?

File: src/app/component/eco-news/components/news-list/news-list.component.ts [hunk 2]
```diff
@@ -55,46 +54,43 @@ export class NewsListComponent implements OnInit, OnDestroy {
...
-      this.allEcoNews.length - this.elements.length;
-
-    this.allEcoNews.forEach((element, index) => {
-      if (index >= this.iterator && index - this.iterator < loadingLength) {
-        this.elements[index] = element;
-      }
-    });
-
-    this.iterator = this.elements.length;
+    if (this.gridOutput) {
```
Comment (mplesha) [inline src/app/component/eco-news/components/news-list/news-list.component.ts:62]: Rewrite it in form:
this.ecoNewsSubscription = this.gridOutput ?
   ... :
   ...

File: src/app/component/eco-news/components/news-list/news-list.component.ts [hunk 3]
```diff
@@ -55,46 +54,43 @@ export class NewsListComponent implements OnInit, OnDestroy {
...
+        .subscribe(this.setList.bind(this));
+    }
   }
 
-  private setAllAndStartingElems(data: EcoNewsModel[]): void {
-    this.allEcoNews = [...data];
-    this.elements = data.splice(0, 12);
-    this.iterator = this.elements.length;
-    this.remaining = this.allEcoNews.length;
+  public setList(data): void {
```
Comment (mplesha) [inline src/app/component/eco-news/components/news-list/news-list.component.ts:71]: what is a type of 'data'?

File: src/app/component/eco-news/components/news-list/news-list.component.ts [hunk 4]
```diff
@@ -55,46 +54,43 @@ export class NewsListComponent implements OnInit, OnDestroy {
...
+    this.remaining = data.totalElements;
```
Comment (mplesha) [inline src/app/component/eco-news/components/news-list/news-list.component.ts:72]: There’s a risk that it's a mutable assignment!

Comment (RomanDeveloperAcc) [reply src/app/component/eco-news/components/news-list/news-list.component.ts:72]: > There’s a risk that it's a mutable assignment!

data.totalElements is a number, not an array which is vulnerable to it

File: src/app/component/eco-news/components/news-list/news-list.component.ts (hunk 1 above)
Comment (RomanDeveloperAcc) [reply src/app/component/eco-news/components/news-list/news-list.component.ts:64]: > The method 'bind' hasn't been used in Angular for a long time. Please check - is it really necessary?

It is necessary to pass the context to setList function

File: src/app/component/eco-news/components/news-list/news-list.component.ts (hunk 2 above)
Comment (RomanDeveloperAcc) [reply src/app/component/eco-news/components/news-list/news-list.component.ts]: > Rewrite it in form:
> this.ecoNewsSubscription = this.gridOutput ?
> ... :
> ...

Fixed

File: src/app/component/eco-news/components/news-list/news-list.component.ts (hunk 3 above)
Comment (RomanDeveloperAcc) [reply src/app/component/eco-news/components/news-list/news-list.component.ts]: > what is a type of 'data'?

Fixed

**Pull request merged**